File: src/main/java/com/skidders/sigma/events/impl/Render3DEvent.java
package com.skidders.sigma.events.impl;

import com.skidders.sigma.events.Event;
import lombok.RequiredArgsConstructor;
import net.minecraft.client.util.math.MatrixStack;

@RequiredArgsConstructor
public class Render3DEvent extends Event {
    public final MatrixStack matrixStack;
    public final float tickDelta;
    public final long limitTime;
}

File: src/main/kotlin/com/skidders/sigma/events/impl/Render3DEvent.kt
package com.skidders.sigma.events.impl;

import com.skidders.sigma.events.Event;
import lombok.RequiredArgsConstructor;
import net.minecraft.client.util.math.MatrixStack;

@RequiredArgsConstructor
data class Render3DEvent(val matrixStack: MatrixStack, val tickDelta: Float, val limitTime: Long) : Event() {
}
